src/replication/document-filter/document-filter.service.spec.ts

```diff
@@ -2,161 +2,148 @@ import { Test, TestingModule } from '@nestjs/testing';
+      _revisions: { start: 1, ids: ['someRev'] },
+      someProperty: 'someValue',
+    };
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
 
-  it('should filter out docs without permissions in BulkGet', () => {
-    const bulkGetResponse: BulkGetResponse = {
-      results: [
-        {
-          id: 'Child:1',
-          docs: [
-            {
-              ok: {
-                _id: 'Child:1',
-                _rev: 'someRev',
-                _revisions: { start: 1, ids: ['someRev'] },
-                someProperty: 'someValue',
-              },
-            },
-          ],
-        },
-        {
-          id: 'School:1',
-          docs: [
-            {
-              ok: {
-                _id: 'School:1',
-                _rev: 'anotherRev',
-                _revisions: { start: 1, ids: ['anotherRev'] },
-                anotherProperty: 'anotherValue',
-              },
-            },
-          ],
-        },
-      ],
-    };
-    service.accessControlList = [{ entity: 'Child', roles: ['admin'] }];
+  it('should filter out docs without read permissions in BulkGet', () => {
+    const bulkGetResponse = createBulkGetResponse(schoolDoc, childDoc);
+    jest.spyOn(mockRulesService, 'getRulesForUser').mockReturnValue([
+      { action: 'write', subject: 'Child' },
+      { action: 'read', subject: 'School' },
+    ]);
 
-    const result = service.transformBulkGetResponse(bulkGetResponse, ['user']);
+    const result = service.filterBulkGetResponse(bulkGetResponse, normalUser);
 
-    expect(result).toEqual({
-      results: [
-        {
-          id: 'School:1',
-          docs: [
-            {
-              ok: {
-                _id: 'School:1',
-                _rev: 'anotherRev',
-                _revisions: { start: 1, ids: ['anotherRev'] },
-                anotherProperty: 'anotherValue',
-              },
-            },
-          ],
-        },
-      ],
-    });
+    expect(result).toEqual(createBulkGetResponse(schoolDoc));
   });
 
-  it('should filter out docs without permissions in AllDocs', () => {
-    const allDocsResponse: AllDocsResponse = {
-      total_rows: 2,
-      offset: 0,
-      rows: [
-        {
-          id: 'Child:1',
-          key: 'someKey',
-          value: { rev: 'someRev' },
-          doc: {
-            _id: 'Child:1',
-            _rev: 'someRev',
-            _revisions: { start: 1, ids: ['someRev'] },
-            someProperty: 'someValue',
-          },
-        },
-        {
-          id: 'School:1',
-          key: 'anotherKey',
-          value: { rev: 'anotherRev' },
-          doc: {
-            _id: 'School:1',
-            _rev: 'anotherRev',
-            _revisions: { start: 1, ids: ['anotherRev'] },
-            anotherProperty: 'anotherValue',
-          },
-        },
-      ],
-    };
-    service.accessControlList = [{ entity: 'School', roles: ['admin'] }];
+  it('should not filter out deleted documents in bulk get', () => {
```

Comment:
Will this also send additional properties to the fronend? Or just the `_id`, `_rev` and the `_deleted` flag which are not personal data?